Student Name: Marion Osong

Student Pace: DSF-FT12-Hybrid

Instructor Name: Samuel Karu

## **MOVIE COMPANY PROJECT.** ##

### *Box office perfomance analysis for new Movie studio.* ###

Student Name: Marion Osong (Group 2)

Student Pace: DSF-FT12-Hybrid

Instructor Name: Samuel Karu

## 🎬Introduction ##



## 🎯Problem statement ##
"Our company aims to establish a new movie studio but lacks experience in film production. To guide the development of a successful film portfolio, we need to analyze historical box office data to identify patterns and factors that contribute to a film's financial success. This analysis will provide actionable insights to inform strategic decisions on genre selection, casting, budgeting, and marketing, ensuring the creation of films with high potential for profitability."
## Objectives ##
### 1. Undestanding and analyze historical Box Office Data.* ####
The Boxoffice Company is a leading global provider of media, technology, and data solutions for the film industry.They partner with top studios, search platforms, and over 300 exhibition partners to connect with moviegoers through tools for ticketing, marketing, and business intelligence.
- **Goal:** 
To identify patterns that inform future project selections.
To also understand the genres that have historically performed well.
      
### *2. Understanding the dataset.* ###
- **Tasks**
   - Understand the technical terms used in the dataset.
   - Review the shape and type of the data, understand the numerical and categorical columns. 
   - Handle missing values (drop or replace). 
   - Identify and remove duplicates. 
   - Drop unnecessary columns and add new ones as needed<p>

3. Business objectives.
- **Tasks**
- Analyze cost structures of successful films to identify areas for cost optimization.

- Develop models to predict the return on investment (ROI) for different types of films.

### *4. statistics.* ###
- **Trend Analysis**
-

### **DATA LOADING.** ###

This step is crucial for ensuring that data is accessible, consistent, and ready for subsequent analysis or modeling

In [47]:
# import the necessaty libraries for data analysis and visualization.

import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import sqlite3
import seaborn as sns
%matplotlib inline



In [48]:
# load the dataset and display the first 5 rows.
df = pd.read_csv('../zipped-data/bom.movie_gross.csv/bom.movie_gross.csv')
df.head()


title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [49]:
# understanding the dataframe and its structure. (Get column attributes and data types.)

df.info() 


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [50]:
df.columns

Index(['title', 'studio', 'domestic_gross', 'foreign_gross', 'year'], dtype='object')

In [51]:
df.shape

(3387, 5)

In [52]:
df.describe(include='all')

title studio  domestic_gross foreign_gross         year
count        3387   3382    3.359000e+03          2037  3387.000000
unique       3386    257             NaN          1204          NaN
top     Bluebeard    IFC             NaN       1200000          NaN
freq            2    166             NaN            23          NaN
mean          NaN    NaN    2.874585e+07           NaN  2013.958075
std           NaN    NaN    6.698250e+07           NaN     2.478141
min           NaN    NaN    1.000000e+02           NaN  2010.000000
25%           NaN    NaN    1.200000e+05           NaN  2012.000000
50%           NaN    NaN    1.400000e+06           NaN  2014.000000
75%           NaN    NaN    2.790000e+07           NaN  2016.000000
max           NaN    NaN    9.367000e+08           NaN  2018.000000

In [53]:
# Total number of null values in the dataset.
df.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [54]:
# Total number of duplicates in the dataset.
df.duplicated().sum()

0

In [55]:
# Total number of the categorical data in the dataset.
df.describe(include='object')

title studio foreign_gross
count        3387   3382          2037
unique       3386    257          1204
top     Bluebeard    IFC       1200000
freq            2    166            23

In [56]:
# Total number of numerical data in the dataset.
df.describe(include='number')

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

### *Summary of the dataset.* ###

- The dataset has a total number of 5 colums and 3387 rows in total.
- Each row represents movie name.
- The dataset has a number of missing values:
    - The studio column has 5 missing values.
    - The domestic_gross column has 28 missing values.
    - foreign_gross: The foreign box office gross revenue
    - The domestic_gross column has 28 missing values.
- The columns consist of:
    - title: The title of the movie.
    - studio: The studio that produced the movie (with some missing values).
    - domestic_gross: The domestic box office gross revenue (with some missing values).
    - foreign_gross: The foreign box office gross revenue (The column the most missing values).
    - year: The year the movie was released.




In [64]:
# loading the sqlite3 library to create a connection to the database.
conn = sqlite3.connect(r'C:\Users\ADMIN\Documents\box-office-movie-insights\zipped-data\im.db\im.db')
cursor = conn.cursor()

In [ ]:
imdb_basics = pd.read_sql("SELECT * FROM movie_basics", conn)
imdb_basics.head()


In [ ]:
# Load data from the movie_basics table
imdb_basics = pd.read_sql("SELECT * FROM movie_basics", conn)
imdb_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [ ]:
# retrieve data from various tables
movie_basics = pd.read_sql_query("SELECT * FROM movie_basics", conn)
movie_ratings = pd.read_sql_query("SELECT * FROM movie_ratings", conn)
directors = pd.read_sql_query("SELECT * FROM directors", conn)
writers = pd.read_sql_query("SELECT * FROM writers", conn)
persons = pd.read_sql_query("SELECT * FROM persons", conn)

#preview the first few rows of each table
print(movie_basics.head(10)), "\n"
print(movie_ratings.head(10)), "\n"
print(directors.head(10)), "\n"
print(writers.head(10)), "\n"
print(persons.head(10)), "\n"


    movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   
5  tt0111414                      A Thin Life                 A Thin Life   
6  tt0112502                          Bigfoot                     Bigfoot   
7  tt0137204                  Joe Finds Grace             Joe Finds Grace   
8  tt0139613                       O Silêncio                  O Silêncio   
9  tt0144449            Nema aviona za Zagreb       Nema aviona za Zagreb   

   start_year  runtime_minutes                      genres  
0        2013            175.0          Action,Crime,Drama  
1        2019            114.0

(None, '\n')

In [73]:
# Load Bom.movie_gross dataset
box_df = pd.read_csv(r'../zipped-data/bom.movie_gross.csv/bom.movie_gross.csv')

# Connect to IMDB SQLite database
conn = sqlite3.connect(r'C:\Users\ADMIN\Documents\box-office-movie-insights\zipped-data\im.db\im.db')

### **DATA CLEANING.** ###

since we have understood the basic data structure and the data types, we now clean the dataset before we can perform Explorative Data Analysis. 

In [74]:
# Check and drop duplicates in each DataFrame
datasets = {
    "box_df": box_df,
    "movie_basics": movie_basics,
    "movie_ratings": movie_ratings,
    "directors": directors,
    "writers": writers,
    "persons": persons
}

for name, df in datasets.items():
    print(f"Checking duplicates in {name}: {df.duplicated().sum()} duplicates found.")
    datasets[name] = df.drop_duplicates()
    print(f"After dropping duplicates, {name} has shape: {datasets[name].shape}")


Checking duplicates in box_df: 0 duplicates found.
After dropping duplicates, box_df has shape: (3387, 5)
Checking duplicates in movie_basics: 0 duplicates found.
After dropping duplicates, movie_basics has shape: (146144, 6)
Checking duplicates in movie_ratings: 0 duplicates found.
After dropping duplicates, movie_ratings has shape: (73856, 3)
Checking duplicates in directors: 127639 duplicates found.
After dropping duplicates, directors has shape: (163535, 2)
Checking duplicates in writers: 77521 duplicates found.
After dropping duplicates, writers has shape: (178352, 2)
Checking duplicates in persons: 0 duplicates found.
After dropping duplicates, persons has shape: (606648, 5)


In [75]:
# handling missing values in the datasets.
for name, df in datasets.items():
    print(f"Missing values in {name}:")
    print(df.isnull().sum())
    print("\n")
    # Drop rows with missing values
    datasets[name] = df.dropna()
    print(f"After dropping missing values, {name} has shape: {datasets[name].shape}")
    

Missing values in box_df:
title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64


After dropping missing values, box_df has shape: (2007, 5)
Missing values in movie_basics:
movie_id               0
primary_title          0
original_title        21
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64


After dropping missing values, movie_basics has shape: (112232, 6)
Missing values in movie_ratings:
movie_id         0
averagerating    0
numvotes         0
dtype: int64


After dropping missing values, movie_ratings has shape: (73856, 3)
Missing values in directors:
movie_id     0
person_id    0
dtype: int64


After dropping missing values, directors has shape: (163535, 2)
Missing values in writers:
movie_id     0
person_id    0
dtype: int64


After dropping missing values, writers has shape: (178352, 2)
Missing values in persons:
person_id                  0
primary_name         

### *Dropping rows and columns with null values.* ###